src/main/java/ch/ivyteam/ivy/maven/AbstractProjectCompileMojo.java

```diff
@@ -42,9 +41,6 @@
 
 public abstract class AbstractProjectCompileMojo extends AbstractEngineMojo
 {
-  @Parameter(property = "project", required = true, readonly = true)
```

Comment:
I think the compile/test goal should not be affected by this change. the provided engine is being manipulated between the start-test-engine and stop-test-engine goal. There an isolated engine can be handy.... here for compilation I do not see much value: but lot's of potential unnecessary engine copying

Reply:
the longer I think about it : the whole flag and its processing should only be in the start/stop goals. There we could even warn users that testing against a cached engine is a bad choice (or turn the feature of in there completely...)

Reply:
another positive thing if isolated in the integration tests goals: we can conveniently write a unit test for this case....

Reply:
in hindsight it does not make much sense to provide this feature for compiling...

I'll have another look at the start/stop test engine mojos

Thank you for the quick feedback